fawltydeps/settings.py

```diff
@@ -232,15 +232,32 @@ def _option_to_toml(name, value) -> str:  # type: ignore
         Options that are of dictionary type must be given a section entry.
         Assumption: dictionaries options are not nested.
         """
-        comment_char = "# " if has_default_value[name] else ""
+        toml_value_map = {
+            "None": "...",  # always commented, see sanity check below
+            "False": "false",
+            "True": "true",
+            # add more values here for which repr(value) is not valid TOML
+        }
+
+        if value is None:
+            # sanity check: None values are represented in TOML by omission,
+            # hence make sure these are always commented (i.e. equal to default)
+            assert has_default_value[name]
+
+        prefix = "# " if has_default_value[name] else ""
+
         if name in dictionary_options:
-            toml_option = f"{comment_char}[tool.fawltydeps.{name}]"
+            toml_option = f"{prefix}[tool.fawltydeps.{name}]"
             if value is not None:
                 toml_option += "".join(
-                    [f"\n{comment_char}{k} = {v!r}" for k, v in value.items()]
+                    [
+                        f"\n{prefix}{k} = {toml_value_map.get(str(v), repr(v))}"
+                        for k, v in value.items()
+                    ]
                 )
         else:
-            toml_option = f"{comment_char}{name} = {value!r}"
+            toml_value = toml_value_map.get(str(value), repr(value))
```

Comment:
These two places where we call `toml_value_map.get(str(value), repr(value))` should ideally be `toml_value_map.get(value, repr(value))` instead (i.e. without the `str(...)`).

However, I suspect there's a good reason for having to string-ify this?

If so, I think it's better to use `repr(...)` instead of `str(...)` here: `toml_value_map.get(repr(value), repr(value))`, as that would map the `--pyenv None` corner-case I discuss above to `"None"` instead of `None`.

(Maybe worth adding a test case for `--pyenv None`?)

Reply:
I could replace `str` by `repr`. but as you have underlined, that would still map `None` to `"None"`, which I understand from your comment above is not the behavior we're expecting.

Reply:
I think it would work because `toml_value_map` contains the `None` string, but it does not contain the _quoted_ `None` string, i.e.:
```python
{
    "None": "...",  # This is what repr(None) maps to
    "'None'": NotImplemented,  # This is what repr("None") maps to...
}
```
Since the second is not in the dict, we would fallback to the default handling, which is what we want for `--pyenv None`.

But this is all irrelevant if we're instead going for the try-except solution above.

Reply:
Replaced `str` with `repr` in [2790667](https://github.com/tweag/FawltyDeps/pull/300/commits/2790667c6fce15b40584d6dd646bac5429ba30a4) as per our discussion.